internal/mesh/internal/controllers/sidecarproxy/mapper.go

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package sidecarproxy
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/controller/cache"
+	"github.com/hashicorp/consul/internal/controller/cache/indexers"
+	"github.com/hashicorp/consul/internal/mesh/internal/meshindexes"
+	"github.com/hashicorp/consul/internal/mesh/internal/types"
+	"github.com/hashicorp/consul/internal/resource"
+	pbauth "github.com/hashicorp/consul/proto-public/pbauth/v2beta1"
+	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v2beta1"
+	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v2beta1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+const (
+	selectedWorkloadsIndexName = "selected-workloads"
+)
+
+// Cache: reverse CED => Service(destination)
+var computedExplicitDestinationsByServiceIndex = indexers.RefOrIDIndex(
+	"computed-explicit-destinations-by-service",
+	func(ced *types.DecodedComputedExplicitDestinations) []*pbresource.Reference {
+		if len(ced.Data.Destinations) == 0 {
+			return nil
+		}
+
+		out := make([]*pbresource.Reference, 0, len(ced.Data.Destinations))
+		for _, dest := range ced.Data.Destinations {
+			if resource.EqualType(pbcatalog.ServiceType, dest.DestinationRef.Type) {
+				out = append(out, dest.DestinationRef)
+			}
+		}
+		return out
+	},
+)
+
+// Cache: reverse CID => Service(destination)
+var computedImplicitDestinationsByServiceIndex = indexers.RefOrIDIndex(
+	"computed-implicit-destinations-by-service",
+	func(ced *types.DecodedComputedImplicitDestinations) []*pbresource.Reference {
+		if len(ced.Data.Destinations) == 0 {
+			return nil
+		}
+
+		out := make([]*pbresource.Reference, 0, len(ced.Data.Destinations))
+		for _, dest := range ced.Data.Destinations {
+			if resource.EqualType(pbcatalog.ServiceType, dest.ServiceRef.Type) {
```

Comment:
Maybe we should call this `DestinationRef` to make this consistent with explicit destinations?

Reply:
Handled in the controller pr.